### PR TITLE
Feature/connectionmanager

### PR DESCRIPTION
**Pull request by bharatgulati**
Connection Management functionality for Salesforce Marketing Cloud

### CODE REVIEW

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 1]
```diff
@@ -100,8 +117,7 @@ private Schema getPluginSchema() throws IOException {
 
   @Test
   public void testConvertRecord() throws ETSdkException, NoSuchFieldException {
-    MarketingConnectorConfig marketingConnectorConfig1 = new MarketingConnectorConfig
-            ("clientId", "clientSecret", "authEndpoint", "soapEndpoint");
+    MarketingCloudConversion cloudConversion = new MarketingCloudConversion();
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: MarketingCloudConversion -> MarketingCloudUtil

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 2]
```diff
@@ -116,13 +132,12 @@ public void testConvertRecord() throws ETSdkException, NoSuchFieldException {
     Mockito.when(MarketingCloudClient.create("clientId", "clientSecret", "authEndpoint",
             "soapEndpoint")).thenReturn(cloudClient);
     Mockito.when(cloudClient.fetchObjectSchema(SourceObject.MAILING_LIST)).thenReturn(cloudObjectInfo);
-    marketingConnectorConfig1.convertRecord(cloudObjectInfo, builder1, object);
+    cloudConversion.convertRecord(cloudObjectInfo, builder1, object);
   }
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Add the assertions, what you are testing here.

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 3]
```diff
@@ -134,7 +149,7 @@ public void testConvertToValueWRecord() {
             Schema.Field.of("bytes", Schema.of(Schema.LogicalType.TIMESTAMP_MICROS)),
             Schema.Field.of("emailid", Schema.of(Schema.Type.DOUBLE)));
     thrown.expect(IllegalStateException.class);
-    Assert.assertNotNull(marketingConnectorConfig1.convertToValue("store_id", fieldSchema, new HashMap<>
+    Assert.assertNotNull(cloudConversion.convertToValue("store_id", fieldSchema, new HashMap<>
             (1)));
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Pls verify some values here, if it is notNull.

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 4]
```diff
@@ -206,35 +221,31 @@ public void testSchema() throws ETSdkException {
...
-    MarketingConnectorConfig marketingConnectorConfig1 = new MarketingConnectorConfig("clientId",
-            " clientSecret", "authEndPoint", "soapEndPoint");
-    Assert.assertEquals(42.0, marketingConnectorConfig1.convertToDoubleValue("42").doubleValue(), 0.0);
-    Assert.assertEquals(42.0, marketingConnectorConfig1.convertToDoubleValue(42).doubleValue(), 0.0);
-    Assert.assertNull(marketingConnectorConfig1.convertToDoubleValue(""));
+    MarketingCloudConversion cloudConversion = new MarketingCloudConversion();
+    Assert.assertEquals(42.0, cloudConversion.convertToDoubleValue("42").doubleValue(), 0.0);
+    Assert.assertEquals(42.0, cloudConversion.convertToDoubleValue(42).doubleValue(), 0.0);
+    Assert.assertNull(cloudConversion.convertToDoubleValue(""));
   }
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Null checkers should be avoided.

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 5]
```diff
@@ -206,35 +221,31 @@ public void testSchema() throws ETSdkException {
...
-    MarketingConnectorConfig marketingConnectorConfig1 = new MarketingConnectorConfig("clientId",
-            " clientSecret", "authEndPoint", "soapEndPoint");
-    Assert.assertEquals(42, marketingConnectorConfig1.convertToIntegerValue("42").intValue());
-    Assert.assertEquals(42, marketingConnectorConfig1.convertToIntegerValue(42).intValue());
-    Assert.assertNull(marketingConnectorConfig1.convertToIntegerValue(""));
+    MarketingCloudConversion cloudConversion = new MarketingCloudConversion();
+    Assert.assertEquals(42, cloudConversion.convertToIntegerValue("42").intValue());
+    Assert.assertEquals(42, cloudConversion.convertToIntegerValue(42).intValue());
+    Assert.assertNull(cloudConversion.convertToIntegerValue(""));
   }
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Null checkers should be avoided.

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java [hunk 6]
```diff
@@ -206,35 +221,31 @@ public void testSchema() throws ETSdkException {
...
-    MarketingConnectorConfig marketingConnectorConfig1 = new MarketingConnectorConfig("clientId",
-            " clientSecret", "authEndPoint", "soapEndPoint");
-    Assert.assertFalse(marketingConnectorConfig1.convertToBooleanValue("Field Value"));
-    Assert.assertFalse(marketingConnectorConfig1.convertToBooleanValue(42));
-    Assert.assertNull(marketingConnectorConfig1.convertToBooleanValue(""));
+    MarketingCloudConversion cloudConversion = new MarketingCloudConversion();
+    Assert.assertFalse(cloudConversion.convertToBooleanValue("Field Value"));
+    Assert.assertFalse(cloudConversion.convertToBooleanValue(42));
+    Assert.assertNull(cloudConversion.convertToBooleanValue(""));
   }
```
Comment (Shubhangi-cs) [inline src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Null checkers should be avoided.

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java (hunk 1 above)
Comment (bharatgulati) [reply src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Changed it to MarketingCloudUtil

File: src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java (hunk 2 above)
Comment (bharatgulati) [reply src/test/java/io/cdap/plugin/sfmc/connector/MarketingConnectorTest.java]: Done

**Pull request closed without merging**